proxy/http/HttpSM.cc

```diff
@@ -4933,8 +4933,10 @@ HttpSM::do_http_server_open(bool raw)
     // between the statement above and the check below.
     // If this happens, we might go over the max by 1 but this is ok.
     if (sum >= t_state.http_config_param->server_max_connections) {
-      ink_assert(pending_action == nullptr);
-      pending_action = eventProcessor.schedule_in(this, HRTIME_MSECONDS(100));
+      // Eventually may want to have a queue as the origin_max_connection does to allow for a combination
+      // of retries and errors.  But at this point, we are just going to allow the error case.
+      t_state.current.state = HttpTransact::CONNECTION_ERROR;
+      call_transact_and_set_next_state(HttpTransact::HandleResponse);
       httpSessionManager.purge_keepalives();
```

Comment:
This is the order we are currently running in Yahoo production.  The purge_keepalives is just clearing out the global pool, so it should not interfere is the set next state.  This was pre-existing code.  I'm guessing the rationale is that is the system is loaded, we should be shutting down idle connections.  Could do the purge first I suppose.